http/route.ts

```diff
@@ -0,0 +1,152 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+
+/**
+ * Request handler for {@linkcode route}.
+ *
+ * > [!WARNING]
+ * > **UNSTABLE**: New API, yet to be vetted.
+ *
+ * @experimental
+ *
+ * Extends {@linkcode Deno.ServeHandlerInfo} by adding making `info` optional
+ * and adding a `params` argument.
+ *
+ * @param request Request
+ * @param info Request info
+ * @param params URL pattern result
+ */
+export type Handler = (
+  request: Request,
+  info?: Deno.ServeHandlerInfo,
+  params?: URLPatternResult | null,
+) => Response | Promise<Response>;
```

Comment:
How about having `StaticRouteHandler` and `DynamicRouteHandler`? `StaticRouteHandler` would have only 2 args, and `DynamicRouteHandler` would have 3